base/range.jl

```diff
@@ -58,6 +50,21 @@ function steprange_last{T}(start::T, step, stop)
     last
 end
 
+function steprange_last_empty{T<:Integer}(start::T, step, stop)
+    # empty range has a special representation where stop = start-1
+    # this is needed to avoid the wrap-around that can happen computing
+    # start - step, which leads to a range that looks very large instead
+    # of empty.
+    if step > zero(step)
+        last = start - one(stop-start)
+    else
+        last = start + one(stop-start)
+    end
+    last
+end
+# For types where x+one(x) may not be well-defined
+steprange_last_empty(start, step, stop) = start - step
```

Comment:
As far as the implementation of this function goes, if you get here, you already know that `step` is in the opposite direction of `zero` as that between `start` and `stop`---see line 33 above. So if you make use of `step`, I don't think you need the branch. But please check!

Or are you proposing this logic as a more general way to detect that the range is empty, without needing even `zero`?